NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -47,10 +47,11 @@ public VehicleModel BuildVehicleModelFrom(GameObject gameObject, TechType techTy
         {
             if (VehicleHelper.IsVehicle(techType))
             {
+                UnityEngine.Component.DestroyImmediate(gameObject.GetComponent<NitroxEntity>()); // template has pre-existing NitroxEntity, remove.
+                NitroxId constructedObjectId = NitroxEntity.GetId(gameObject);
```

Comment:
This is something that was started with #1908 but we shouldn't use GetId as an ID setter anymore (it will attach a random ID when it's not existing, which not something we want in most of the cases)

Reply:
Is saw this too but NitroxEntity has no correct method for it right now. I will just update it with #1908.

Reply:
We should talk a bit about that.  Throwing when it doesn't exist is very rigid.  It is what we used to do in a lot of places (like RequireComponent or Validate.NotNull) but in the end it causes functionality to bomb out entirely instead of operate in a semi-degraded state.